modules/apps/transfer/keeper/relay.go

```diff
@@ -109,6 +109,12 @@ func (k Keeper) sendTransfer(
 		); err != nil {
 			return 0, err
 		}
+
+		// get the existing amount in escrow.
+		existingAmount := k.bankKeeper.GetBalance(ctx, escrowAddress, token.Denom)
+
+		// store the token about to be IBC'd Out here
+		k.SetIBCOutDenomAmount(ctx, token.GetDenom(), existingAmount.Amount)
```

Comment:
Maybe I'm missing something, but I think that this is setting the total amount escrowed for `token.Denom` on a _particular_ channel, not across all channels, right? In the example you mention in the description of the PR, I think this code would set the total amount in escrow in `channel-1`, `channel-2` or `channel-3` depending on what channel the transfer is going out at any given time. For example, if I do a transfer on `channel-1` it sets the total amount in escrow for `channel-1`, but if afterwards I do a transfer on `channel-2` then the previous vale would get overwritten with the total amount in escrow for `channel-2`...

Reply:
aah i see, i went with the assumption that all channels share same escrowAccount. So here, i'll probably have to **get all the channel's escrow account sum it up and then store in every transfer** right?

Reply:
Also i was thinking, are all transfers happening under "channels in one port"? Do i also have check for different ports that contains different channels for transfers?

Reply:
Yes, exactly. Tokens can be transferred out of a chain to the same destination chain through different channels, so you'll have add up the the balances of potentially multiple escrow accounts.

Reply:
sounds great. I have addressed most of it as per this [comment](https://github.com/cosmos/ibc-go/pull/3019#issuecomment-1399208014)